code/datums/components/bad_leg.dm

```diff
@@ -0,0 +1,146 @@
+//not variables to save on overhead
+
+#define MESSAGE_COOLDOWN 1.5 SECONDS
+
+/datum/component/bad_leg
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// Maximum amount of steps you can take until you get stunned from pain.
+	var/max_steps = 30
+	/// Tracker for steps walked.
+	var/steps_walking
+	/// Holds the last time a message was sent to prevent spam.
+	var/last_message_time
+	/// The bound action to reduce steps tracker.
+	var/datum/action/human_action/rest_legs/bound_action
+	/// Bound human.
+	var/mob/living/carbon/human/parent_human
```

Comment:
register comsig_parent_qdel to a cleanup proc on this + the limb to prevent harddels